environment/src/main/java/com/sequenceiq/environment/environment/v1/converter/EnvironmentApiConverter.java

```diff
@@ -178,16 +177,18 @@ private ParametersDto azureParamsToParametersDto(AzureEnvironmentParameters azur
     }
 
     private AwsParametersDto awsParamsToAwsParameters(AwsEnvironmentParameters aws, AwsFreeIpaParameters awsFreeIpa) {
-        return AwsParametersDto.builder()
+        AwsParametersDto.Builder builder = AwsParametersDto.builder()
                 .withDynamoDbTableName(Optional.ofNullable(aws)
                         .map(AwsEnvironmentParameters::getS3guard)
                         .map(S3GuardRequestParameters::getDynamoDbTableName)
-                        .orElse(null))
-                .withFreeIpaSpotPercentage(Optional.ofNullable(awsFreeIpa)
-                        .map(AwsFreeIpaParameters::getSpot)
-                        .map(AwsFreeIpaSpotParameters::getPercentage)
-                        .orElse(0))
-                .build();
+                        .orElse(null));
+        Optional.ofNullable(awsFreeIpa)
+                .map(AwsFreeIpaParameters::getSpot)
+                .ifPresent(awsFreeIpaSpotParameters -> {
+                    builder.withFreeIpaSpotPercentage(awsFreeIpaSpotParameters.getPercentage())
```

Comment:
what if spotPercentage is not present? the orElse(0) logic is changed here

Reply:
null is treated as a default 0 value, so it was not necessary in the first place